### PR TITLE
Add Test Coverage

### LINKED ISSUE #88: Need a negative test for invalid user name or password for Basic authentication
we need a negative test that catches exactly the kind of exception we expect for a service that throws SecurityTokenException for an invalid user name or password. A negative test that validates the ArgumentNullException path would also be good to include.

### PR DESCRIPTION
**Pull request by hongdai**
* Add a test to cover incorrect password.
* Add a test to cover empty user name

Fix #88

### CODE REVIEW

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs [hunk 1]
```diff
@@ -54,6 +55,59 @@ public static void BasicAuthentication_RoundTrips_Echo()
     }
 
     [Fact]
+    [OuterLoop]
+    public static void BasicAuthenticationInvalidPwd_throw_MessageSecurityException()
+    {
+        StringBuilder errorBuilder = new StringBuilder();
+        //Will need to use localized string once it is available
+        string message = "The HTTP request was forbidden with client authentication scheme 'Basic'.";
```
Comment (hongdai) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: It turns out the exception message has stripped in .Net Native retail build. I will update it to take it into consideration.

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs [hunk 2]
```diff
@@ -54,6 +55,62 @@ public static void BasicAuthentication_RoundTrips_Echo()
...
+
+    [Fact]
+    [ActiveIssue(224)]
+    [OuterLoop]
+    public static void BasicAuthenticationEmptyUser_throw_ArgumentException()
+    {
+        StringBuilder errorBuilder = new StringBuilder();
+        //Will need to use localized string once it is available
+        string message = "The username cannot be empty.";
+
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: I think we will run into trouble with NET Native retail changing the message.
We should verify the ArgumentException.ParamName is correct for any kind of ArgumentException.
There is a special overload of Assert.Throws that takes this name -- look in CreateChannel_ExpectedNetTcpScheme_HttpScheme_ThrowsUriFormat

Comment (hongdai) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: I've just verified the ParamName in this scenario s NULL on Desktop.

I agree we will run into with NET Native retail changing the message. However it seems we do not have better option right now.

Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: My point here was that this test is guaranteed to fail on NET Native because you are comparing the full message.  You need to find a way to make this test pass on NET Native (when it is enabled), either by finding some substring to match or not testing the message.

Comment (hongdai) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Https/ClientCredentialTypeTests.cs]: Point taken. When issue #224 get fixed, I will find some substring and update the test.

**Comment by roncain** [comment]
LGTM.  For tests, I think we need only one approver, so go ahead and merge when you want.
